lib/NonlinearSolveBase/src/linear_solve.jl

```diff
@@ -100,7 +100,7 @@ fix_incompatible_linsolve_arguments(A, b, u) = u
 fix_incompatible_linsolve_arguments(::SArray, ::SArray, u::SArray) = u
 function fix_incompatible_linsolve_arguments(A, b, u::SArray)
     (Core.Compiler.return_type(\, Tuple{typeof(A), typeof(b)}) <: typeof(u)) && return u
-    @warn "Solving Linear System A::$(typeof(A)) x::$(typeof(u)) = b::$(typeof(u)) is not \
+    @warn lazy"Solving Linear System A::$(typeof(A)) x::$(typeof(u)) = b::$(typeof(u)) is not \
```

Comment:
FYI, this is actually the only instance I care about. But I changed all the other occurences since I was already at it.